src/toaster/src/Toaster.js

```diff
@@ -15,15 +16,29 @@ export default class Toaster {
     container.setAttribute('data-evergreen-toaster-container', '')
     document.body.appendChild(container)
 
-    ReactDOM.render(
+    const el = () => (
       <ToastManager
         bindNotify={this._bindNotify}
         bindRemove={this._bindRemove}
         bindGetToasts={this._bindGetToasts}
         bindCloseAll={this._bindCloseAll}
-      />,
-      container
+      />
     )
+
+    if (getMajorVersion(ReactDOM.version) >= 18) {
```

Comment:
This is clever! I hadn't thought to introspect on the package itself to see if we could conditionally choose which API to use for rendering. Have you seen any other libraries taking this same approach to opt-in to the new API depending on the running React version?

Instead of just returning in the `catch` block, how do you feel about falling back to `ReactDOM.render` in the event something throws here? I'm not sure what might cause this block to `throw`, but defensively trying to render using the function that was previously the standard seems better than giving up entirely. We know it should be available to use until at least the next major version of React.

Reply:
I've done this approach sometimes on some projects and they have working for now.

Otherwise, yep you are right, with this we can be more confident that if something happens we will render it with the previous version. 

Thanks for the feedback @brandongregoryscott